frontend/src/components/Form.js

```diff
@@ -61,8 +63,9 @@ const Form = () => {
   const onSubmit = (data) =>
     mutation.mutate(data, {
       onSuccess: (data, variables, context) => {
-        console.log('success!');
-        console.log(data.data);
+        const id = data.data.id;
+        queryClient.setQueryData(['applications', { id: id }], data);
```

Comment:
The id here doesn't have to be in an object it should match above `['application', id]`